File: 2575-find-the-divisibility-array-of-a-string/2575-find-the-divisibility-array-of-a-string.java
class Solution {
    public int[] divisibilityArray(String word, int m) {
        long rem=0;
        int[] res=new int[word.length()];
        for(int i=0;i<word.length();i++){
            rem=(rem*10+(word.charAt(i)-'0'))%m;
            if(rem==0)res[i]=1;
            else res[i]=0;
        }
        return res;
    }
}

File: 2575-find-the-divisibility-array-of-a-string/2575-find-the-divisibility-array-of-a-string.cpp
class Solution {
public:
    vector<int> divisibilityArray(string word, int m) {
        
        long long rem=0;vector<int>res(word.size(),0);
        for(int i=0;i<word.size();i++){

            rem=(10*rem+(word[i]-'0'))%m;
            if(rem==0)res[i]=1;
            
        }
        return res;
            
    }
};